From c549e115fc3b50db082640c7e456e5f952bb9501 Mon Sep 17 00:00:00 2001
From: Nhan Nguyen <nhan.nguyen.yb@renesas.com>
Date: Wed, 14 Nov 2018 15:55:53 +0700
Subject: [PATCH 594/628] Revert "Add capture width check for NV16 format"

This reverts commit db2e671f974e ("Add capture width check for NV16 format")

Signed-off-by: Nhan Nguyen <nhan.nguyen.yb@renesas.com>
---
 drivers/media/platform/soc_camera/rcar_vin.c | 18 +-----------------
 1 file changed, 1 insertion(+), 17 deletions(-)

diff --git a/drivers/media/platform/soc_camera/rcar_vin.c b/drivers/media/platform/soc_camera/rcar_vin.c
index 07f7226..6c89843 100644
--- a/drivers/media/platform/soc_camera/rcar_vin.c
+++ b/drivers/media/platform/soc_camera/rcar_vin.c
@@ -138,12 +138,6 @@
 #define VIN_MAX_WIDTH		2048
 #define VIN_MAX_HEIGHT		2048
 
-/*
- * At the time of NV16 format specification, the value of ow should be
- * specifiedthe multiple of 32. To take the screen width normalizedto 640.
- */
-#define RCA_CAMERA_USING_FORMAT_NV16_WIDTH		640
-
 #define TIMEOUT_MS		100
 
 enum chip_id {
@@ -667,18 +661,8 @@ static int rcar_vin_setup(struct rcar_vin_priv *priv)
 	/* output format */
 	switch (icd->current_fmt->host_fmt->fourcc) {
 	case V4L2_PIX_FMT_NV16:
-	/*
-	 * At the time of NV16 format specification, the capture output width should be
-	 * specified the value of the multiple of 32 for the specification of H/W.
-	 * In addition, the vertical scaling function by NV16 format is forbidden.
-	 */
-		if (cam->width%32 != 0) {
-			iowrite32(ALIGN(RCA_CAMERA_USING_FORMAT_NV16_WIDTH * cam->height, 0x80),
-			  priv->base + VNUVAOF_REG);
-		} else {
-			iowrite32(ALIGN(cam->width * cam->height, 0x80),
+		iowrite32(ALIGN(cam->width * cam->height, 0x80),
 			  priv->base + VNUVAOF_REG);
-		}
 		dmr = VNDMR_DTMD_YCSEP;
 		output_is_yuv = true;
 		break;
-- 
2.7.4

